### PR TITLE
PIMS-1802: Bulk Upload Page

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

  
[PIMS-1802](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-1802)

* Adds a bulk upload page with similar elements to the PIMS v1 version.
* Can be navigated to from the admin menu in the header. As such, it is a protected route.
* You should be able to click the dashed line area or drag and drop into it to attach a file.
* Once a file is attached, you can hit upload and the file will begin processing.
* The loading bar will progress based on the contents of the API call made every X seconds using the interval inside the useEffect.
* You can download the results of your most recent upload as an XLSX file or just view in the MUI data grid component.
* There are also some small changes to the backend. Mostly just adding some missed fields into the import service and modifying the way that the import_result row is first created so that it can be returned in the initial call the to the import endpoint.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

**Review by dbarkowsky — COMMENTED** [review]
Like the UI here. 

- Land Area doesn't seem to be populated, even when the row has a value. Makes the details page look like this:
image
- PID/PIN may have to be joint optional fields. Right now PID is mandatory, but there are parcels that will only have a PIN.

**Comment by GrahamS-Quartech** [comment]
> Like the UI here.
> 
>     * Land Area doesn't seem to be populated, even when the row has a value. Makes the details page look like this:

I think this might just be the test CSV having the wrong header (should be LandArea and not Land (space) Area).

File: express-api/src/services/properties/propertiesServices.ts [hunk 1]
```diff
@@ -447,7 +448,7 @@ const importPropertiesAsJSON = async (
   // let queuedParcels = [];
   // let queuedBuildings = [];
   const queryRunner = AppDataSource.createQueryRunner();
-  await queryRunner.startTransaction();
+  // await queryRunner.startTransaction();
```
Comment (dbarkowsky) [inline express-api/src/services/properties/propertiesServices.ts:451]:
```suggestion
```

File: express-api/src/services/properties/propertiesServices.ts [hunk 2]
```diff
@@ -508,7 +509,7 @@ const importPropertiesAsJSON = async (
     logger.warn(e.message);
     logger.warn(e.stack);
   } finally {
-    await queryRunner.rollbackTransaction(); //NOTE: This rollback provided for testing convenience. Will be removed for final merge.
+    // await queryRunner.rollbackTransaction(); //NOTE: This rollback provided for testing convenience. Will be removed for final merge.
```
Comment (dbarkowsky) [inline express-api/src/services/properties/propertiesServices.ts:512]:
```suggestion
```

**Review by dbarkowsky — COMMENTED** [review]
Working well for me. Knocking it out of the park. ⚾ 
A few comments to discuss then let's get it in.

**Comment by GrahamS-Quartech** [comment]
> PID/PIN may have to be joint optional fields. Right now PID is mandatory, but there are parcels that will only have a PIN.

Just noting here that Steven helped us decide that PID will remain mandatory for now.

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**